src/Model/Shopgate/Extended/Base.php

```diff
@@ -105,8 +105,7 @@ public function setItems($list)
 
             $item = $this->itemFactory->create();
             $item->loadArray($element);
-            $itemIds = explode('-', $item->getItemNumber());
-            $items[array_pop($itemIds)] = $item;
+            $items[$item->getItemNumber()] = $item;
```

Comment:
will this still work if itemNumber is  in [parent]-[child] format?

Reply:
yes, because these items are referenced in the request with [parent]-[child] too.
I tested check_cart and add_order multiple times with configurable and simple products.